### DISCUSSION TITLE
TOTP QR code validation and other methods without emailing

### DISCUSSION THREAD

**Discussion by boristhefirst** [authelia/authelia#6074] [Q&A] [answered]
I would like to test protection of email logon page with Authelia.
Is it possible to be configured validation of totp QR code (and the other method of course like Webauthn security key) without emailing link to users email addresses like this one `https://auth.example.com/one-time-password/register?token=eyJhbGciOiJIUzI1N...`.
I need to test that because if two-factor is required for protected email logon page the users can't login to check the same mailbox during the validation of the chosen method and requirement for two factor in the Authelia configuration.
That what I mean is the following: Is it possible when users are signing in for a first time through Authelia protected email logon page with two factor required and choose totp (or webauthn) to view first directly the QR code (webauth key pairing message) and to validate it without any visible option to send link via email with validation url to the email address of the user?

I've changed the listening port of Authelia from 9091 to 443 if that matters.
And one other issue appeared. I've tried to use the the authenticator extension of Chrome browser to scan the QR for further generation of one-time-passwords and every time when try the logon is failing with message `The one-time password might be wrong`. But this is not important for me. I'm just testing it.  If needed I would use recommended authenticators.

In connection with that I've get into another issue. I've tried to delete the TOTP configuration for this user and to validate again new QR code.
I'm using mysql (mariadb) storage and AD for authentication backend. When I use this CLI
```
authelia storage user totp delete username --encryption-key Very-Long-Ecryption-Key --mysql.database authelia --mysql.username authelia --mysql.password P@ssw0rd
``` 
happens that `-bash: !Sy: event not found`.

**Comment by james-d-elliott** [comment, marked as answer]
You can generate TOTP configurations via the CLI and save them as a PNG. We have no plans to add any other method at this time unless a reasonably safe option comes to light. We could consider disabling the requirement but I feel like that may add complexity which could reasonably lead to security issues if not done right so I'd prefer to avoid it.

The issue with the encryption key is probably it has a shell special character, try using single quotes like this `'Very-Long-Ecryption-Key'`.

**Reply by boristhefirst** [reply]
Thank you @james-d-elliott 

I've corrected my authelia CLIs adding quotes and it works.
In case of protecting email login portal this suggestion with TOTP generation is one of the paths. There is other possibilities too because email systems allow other type of access and delivery protocols so it depends from well done deployment plan and your answer suggested me about that too.

I'm going further testing Authelia but I've blocked to one of the next steps.
Before to go deeper in OIDC I need to clear something about ldap authentication. I know that it's not the right place in this discussion.

I'm using `implementation: activedirectory` and the following settings `base_dn: DC=example,DC=inter-medial-domain-name,DC=com`  , `username_attribute: sAMAccountName` and
 `users_filter: (&(|({username_attribute}={input})({mail_attribute}={input}))(sAMAccountType=8053...` 

Is it possible to use logon name like `domain\username` (down-level logon name) for Authelia logon page in the first factor and Active Directory ldap background?

I mean something like example\john.smith  where john.smith is sAMAccountName .

I've tried to modify `username_attribute: dn\sAMAccountName` `username_attribute: domain\sAMAccountName` `username_attribute: dc\sAMAccountName` and many others.

I've tried also modification of  `users_filter: (&(|({domain}\{username_attribute}={input})({mail_attrib...` modifying with this **{domain}\\{username_attribute}={input}** and various similar expressions but it not works.

Edit: I've seen interesting syntax for attributes. I didn't met it before using 7.37.5 Bare-Metal

```
  attributes:
    distinguished_name: 'distinguishedName'
    username: 'uid'
    display_name: 'displayName'
    mail: 'mail'
    member_of: 'memberOf'
    group_name: 'cn'
``` 

`distinguished_name: 'distinguishedName'` is new for me. So I believe that other not known attributes are existing!
